### PR TITLE
Fix long connection times.

### PR DESCRIPTION
**Pull request by SebastianGrans**
A more elegant solution than PR #4.

`StateConnection.start()` is now a blocking call until one of each message type has been received. This also has the effect that `Moto(start_state_connection=True)` is blocking. 

This prevents the user from trying to access parameters, such as `StateConnection.robot_status()` or `Moto.state.robot_status()`, before they have been instantiated.

### CODE REVIEW

File: moto/state_connection.py [hunk 1]
```diff
@@ -65,9 +66,13 @@ def add_joint_feedback_msg_callback(self, callback: Callable):
     def add_joint_feedback_ex_msg_callback(self, callback: Callable):
         self._joint_feedback_ex_callbacks.append(callback)
 
-    def start(self) -> None:
+    def start(self, timeout=5) -> None:
```
Comment (tingelst) [inline moto/state_connection.py:69]: Add type annotation to timeout argument.

**Comment by tingelst** [comment]
Please format file using the Black code formatter.

File: moto/state_connection.py [hunk 2]
```diff
@@ -66,13 +67,15 @@ def add_joint_feedback_msg_callback(self, callback: Callable):
     def add_joint_feedback_ex_msg_callback(self, callback: Callable):
         self._joint_feedback_ex_callbacks.append(callback)
 
-    def start(self, timeout=5) -> None:
+    def start(self, timeout: Float = 5) -> None:
```
Comment (tingelst) [inline moto/state_connection.py:70]: Nitpicking: 5 -> 5.0

File: moto/state_connection.py [hunk 3]
```diff
@@ -96,11 +99,10 @@ def _worker(self) -> None:
...
-            if not self._initial_response.is_set() and \
-                (self.robot_status is not None and \
-                self._joint_feedback is not None and \
-                self.joint_feedback_ex is not None):
+            if not self._initial_response.is_set() and (
+                self.robot_status is not None
+                and self._joint_feedback is not None
+                and self.joint_feedback_ex is not None
+            ):
                 print("At least one message of every time has been received.")
```
Comment (tingelst) [inline moto/state_connection.py:107]: Might be we should move to use logging and not prints.

Comment (SebastianGrans) [reply moto/state_connection.py:107]: Removing the print statement now. 

I think logging should be implemented everywhere, so I added it to the project board as a to-do. 
[Link](https://github.com/ra-mtp-ntnu/moto/projects/1#card-74143509)

**Pull request merged**